From bc4546204ab13ebc0e1f46ed438a2db7d66af465 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Tue, 25 Jan 2011 17:34:56 +0000
Subject: search: Reflect changes to pattern string

Some dialog options modify the pattern string, make sure to update
our GRegex instance.

Originally found in https://bugzilla.redhat.com/show_bug.cgi?id=669113
---
diff --git a/src/terminal-search-dialog.c b/src/terminal-search-dialog.c
index e299ae9..4a3bf9c 100644
--- a/src/terminal-search-dialog.c
+++ b/src/terminal-search-dialog.c
@@ -365,7 +365,8 @@ terminal_search_dialog_get_regex (GtkWidget *dialog)
       g_free ((char *) old_pattern);
   }
 
-  if (!priv->regex || priv->regex_compile_flags != compile_flags) {
+  if (!priv->regex || priv->regex_compile_flags != compile_flags ||
+      g_strcmp0 (pattern, g_regex_get_pattern (priv->regex)) != 0) {
     priv->regex_compile_flags = compile_flags;
     if (priv->regex)
       g_regex_unref (priv->regex);
--
cgit v0.8.3.1
